libazureinit/src/goalstate.rs

```diff
@@ -80,21 +113,33 @@ pub async fn report_health(
 
     let post_request = build_report_health_file(goalstate);
 
-    let response = client
-        .post(url)
-        .headers(headers)
-        .body(post_request)
-        .send()
-        .await?;
-
-    if response.status().is_success() {
-        Ok(())
-    } else {
-        Err(Error::HttpStatus {
-            endpoint: url.to_owned(),
-            status: response.status(),
-        })
-    }
+    _ = timeout(total_timeout, async {
+        loop {
+            if let Ok(response) = client
+                .post(url)
+                .headers(headers.clone())
+                .body(post_request.clone())
+                .timeout(Duration::from_secs(30))
+                .send()
+                .await
+            {
+                let statuscode = response.status();
+
+                if statuscode.is_success() && statuscode == StatusCode::OK {
+                    return Ok(response);
+                }
+
+                if http::RETRY_CODES.contains(&statuscode) {
```

Comment:
Same here, I assume we want to return only if the status code _isn't_ in the list of retry codes.